ex-sdl-cairo-freetype-harfbuzz.c

```diff
@@ -211,25 +216,27 @@ int main () {
         while(SDL_PollEvent(&event)) {
             switch (event.type) {
                 case SDL_KEYDOWN:           
-                    if (event.key.keysym.sym == SDLK_ESCAPE) {
                         done = 1;
-                    }
                     break;
 
 	        case SDL_QUIT:
                     done = 1;
                     break;
 
-                case SDL_VIDEORESIZE:
-                    width = event.resize.w;
-                    height = event.resize.h;
-                    screen = SDL_SetVideoMode(event.resize.w, event.resize.h, bpp, videoFlags);
-                    if (!screen) {
-                        fprintf(stderr, "Could not get a surface after resize: %s\n", SDL_GetError( ));
-                        exit(-1);
+                case SDL_WINDOWEVENT: {
+			switch (event.window.event) {
+				case SDL_WINDOWEVENT_SIZE_CHANGED:
+					width = event.window.data1;
+					height = event.window.data2;
+					break;
+				case SDL_WINDOWEVENT_CLOSE:
+					event.type = SDL_QUIT;
+					SDL_PushEvent(&event);
+					break;
+				      }
                     }
                     /* Create an SDL image surface we can hand to cairo to draw to */
-                    SDL_FreeSurface(sdl_surface);
+                    //SDL_FreeSurface(sdl_surface);
```

Comment:
We should be freeing this if we are creating it down below, otherwise it's a notable memory leak

Reply:
fixed